base/channels.jl

```diff
@@ -8,9 +8,9 @@ Representation of a channel passing objects of type `T`.
 abstract type AbstractChannel{T} end
 
 """
-    Channel{T}(sz::Int=0)
+    Channel{T=Any}(size::Int=0)
```

Comment:
Wow... maybe we should actually implement this syntax for defining constructors with default type parameter values? Seems quite possible.

Reply:
> Wow... maybe we should actually implement this syntax for defining constructors with default type parameter values? Seems quite possible.

:D Yeah that makes sense to me! It seems pretty readable, and has an intuitive meaning, i think.